cli/src/cli_util.rs

```diff
@@ -1313,6 +1313,17 @@ See https://github.com/martinvonz/jj/blob/main/docs/working-copy.md#stale-workin
                 // update it.
             }
         }
+
+        let current_wc = self.repo().view().wc_commit_ids().get(self.workspace_id());
+        if let Some(wc) = current_wc {
+            if self.check_rewritable([wc]).is_err() {
+                writeln!(
+                    ui.warning_default(),
+                    "@ became immutable. You should create a new commit using jj new."
+                )?;
+                // Here i should create a new commit.
+            }
+        }
```

Comment:
I managed to print this happy message when the problem occurs.

However idk how to create the new commit as i can't use tx anymore and self.repo is readonly. 
I tried creating a new transaction, but i cannot update the working copy commit before finishing the transaction, but finishing the transaction will starts infinite recursion loop as the wc commit has not been updated, the problem will occurr infinitely many times.

Reply:
> However idk how to create the new commit as i can't use tx anymore and self.repo is readonly.

This check should be done before committing the transaction, so before line 1305. You'll want to call `tx.mut_repo().check_out(&maybe_new_wc_commit)` and then update `maybe_new_wc_commit` to be the new commit you get back. HTH